### ISSUE TITLE
diagnose_report() produces output with too much rows in section 1.2.1 Diagnosis of categorical variables

### ISSUE THREAD

**Issue by sedechio** [choonghyunryu/dlookr#34]
The following code produces a html output where the section 1.2.1 Diagnosis of categorical variables contains too many rows.
It appears when there are more than 10 variable values with a frequency from top 10 frequency.
Maybe an additional level of rank() or rownum() should be used.
R version: 4.0.2
dlookr version: 0.4.2
Could you please check/fix?

```{R}
library(dlookr)
#test for diagnose_report() categorical variable part error
    datafrm <- data.frame("field1" = c(paste("f1val", "v01")), "field2"= c(paste("f2val","v02")))
for (row in 1:100) {
    datafrm <- rbind(datafrm, data.frame("field1" = c(paste("f1val", row)), "field2"= c(paste("f2val",row))))
    
}
    
for (row in 1:100) {
    datafrm <- rbind(datafrm, data.frame("field1" = c(paste("f1val", row)), "field2"= c(paste("f2val",row))))
    
}
# this output does not look good:
  diagnose_report(datafrm, output_format = "html", output_file = "testdiagnose_bad.html", output_dir = "c:/temp")
    
for (row in 1:5) {
    datafrm <- rbind(datafrm, data.frame("field1" = c(paste("f1val", row)), "field2"= c(paste("f2val",row))))
    
}
  # this output looks good 
diagnose_report(datafrm, output_format = "html", output_file = "testdiagnose_good.html", output_dir = "c:/temp")
```

**Comment by choonghyunryu** [comment]
@sedechio,

Thank you for your issue feedback. I'll fix that issue in the next 0.4.4 version.

**Comment by choonghyunryu** [comment]
@sedechio,

This issue is caused by the diagnose_category() function.
First, there was an error in the rank variable in the data frame object returned by the diagnose_category() function. dlookr used to calculate the row number, not the rank for the frequency. This was modified so that the correct rank is displayed.

The diagnose_category() function returns the levels of the top 10 rank, unless the argument is changed.
Your data case has the same frequency for each level, so all levels have a rank of 1. Therefore, all levels have been returned.

So, I added the type argument to the diagnose_category() function. "rank" returns the levels corresponding to the top n ranks, and "n" returns only the top n levels.

So, I set the type argument of the diagnose_category() function to "n" in the diagnose_report() function. It only prints the top 10 levels.

This feature is currently implemented in the development version of github. I'll add the 0.4.4 version to CRAN.
